### ISSUE TITLE
LineChart not being populated in ResponsiveContainer, no error messages

### ISSUE THREAD

**Issue by paddymul** [recharts/recharts#5350]
## Do you want to request a feature or report a bug?
This is a bug report

## What is the current behavior?
I call the rechart LineChart example
```
export class RechartExample extends PureComponent {
  static demoUrl = 'https://codesandbox.io/p/sandbox/line-chart-width-xaxis-padding-8v7952';

  render() {
    console.log("rechart example");
    return (
      <ResponsiveContainer width="100%" height="100%">
        <LineChart
          width={500}
          height={300}
```
I can see the Responsive Container in the dom, but LineChart isn't present.  I can also see the console.log of "rechart example"

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via codesandbox or similar (template: https://codesandbox.io/p/sandbox/simple-line-chart-v25dl4).

This is probably a packaging problem.  I'm working to make recharts work the esbuild based build tooling of anywidget.

`RechartExample` is here https://github.com/paddymul/anywidget-react-starter/blob/feat/failing-rechart-render/packages/some-react-tsxlib/src/SampleComponent.tsx

That is invoked here https://github.com/paddymul/anywidget-react-starter/blob/feat/failing-rechart-render/js/widget.tsx#L18
AnyWidget compiles this via esbuild.

## What is the expected behavior?

I would expect the DOM to be populated.  I would also expect some type of error for a failure, not just a silent failure

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
`"recharts": "^2.13.1"`

This is almost certainly a pckaging issue, but it is hard to diagnose.

**Comment by ckifer** [comment]
When `height` and `width` are percentages, your parent containers also must a height/width. Do things render when you set height to a static value?

**Comment by ckifer** [comment]
https://codesandbox.io/p/sandbox/line-chart-width-xaxis-padding-forked-34skky

**Comment by paddymul** [comment]
Yep, that fixed it.  An error message here would be really helpful.  When I get some time, I will try to write a PR to add one.

**Comment by ckifer** [comment]
https://github.com/recharts/recharts/blob/3.x/src/component/ResponsiveContainer.tsx#L145-L158 does this warning not fire? if so when you look can you see why it isn't firing?

A lot of people don't like the warnings that are in responsive container so I would be extremely hesitant about adding an error (which is also a breaking change)

**Comment by ckifer** [comment]
Going to close this as stale, if anyone wants to PR for this that would be fine
